src/components/charts/SwipeChart.tsx

```diff
@@ -364,39 +387,46 @@ const SwipeChartComponent = (params: Props) => {
     )
   })
 
-  const handleRenderTimespanButton = useHandler((label: string, timespanKey: Timespan) => (
+  const renderTimespanButton = useHandler((label: string, timespanKey: Timespan) => (
     <MinimalButton
       key={timespanKey}
-      marginRem={[0, 0.5, 0, 0.5]}
+      marginRem={BUTTON_MARGINS}
       label={label}
       highlighted={selectedTimespan === timespanKey}
-      onPress={() => {
-        setSelectedTimespan(timespanKey)
-      }}
+      onPress={handleSetSelectedTimespan(timespanKey)}
```

Comment:
This memoization doesn't help as you are calling the function, not passing it to `onPress`. This can only be solved by creating memoized useHandlers for each timespan. ie.

```
handleSetTimespan1D()
handleSetTimespan1W()
etc...
```